pkg/controller/perconaservermongodbrestore/physical.go

```diff
@@ -297,107 +496,150 @@ func (r *ReconcilePerconaServerMongoDBRestore) prepareStatefulSetsForPhysicalRes
-			nn := types.NamespacedName{Namespace: cluster.Namespace, Name: stsName}
-			err := r.client.Get(ctx, nn, &sts)
-			if err != nil {
-				return err
-			}
+			return r.updateStatefulSetForPhysicalRestore(ctx, cluster, types.NamespacedName{Namespace: cluster.Namespace, Name: stsName})
+		})
+		if err != nil {
+			return errors.Wrapf(err, "prepare statefulset %s for physical restore", stsName)
+		}
+	}
 
-			// Annotating statefulset to stop reconciliation in psmdb_controller
-			sts.Annotations[psmdbv1.AnnotationRestoreInProgress] = "true"
+	return nil
+}
 
-			cmd := []string{
-				"bash", "-c",
-				"install -D /usr/bin/pbm /opt/percona/pbm && install -D /usr/bin/pbm-agent /opt/percona/pbm-agent",
-			}
-			pbmInit := psmdb.EntrypointInitContainer(
-				cluster,
-				"pbm-init",
-				cluster.Spec.Backup.Image,
-				cluster.Spec.ImagePullPolicy,
-				cmd,
-			)
-			sts.Spec.Template.Spec.InitContainers = append(sts.Spec.Template.Spec.InitContainers, pbmInit)
-
-			// remove backup-agent container
-			pbmIdx := -1
-			for idx, c := range sts.Spec.Template.Spec.Containers {
-				if c.Name == "backup-agent" {
-					pbmIdx = idx
-					break
-				}
-			}
-			if pbmIdx == -1 {
-				return errors.New("failed to find backup-agent container")
+func (r *ReconcilePerconaServerMongoDBRestore) getUserCredentials(ctx context.Context, cluster *psmdbv1.PerconaServerMongoDB, role psmdbv1.UserRole) (psmdb.Credentials, error) {
+	creds := psmdb.Credentials{}
+
+	usersSecret := corev1.Secret{}
+	err := r.client.Get(ctx, types.NamespacedName{Name: psmdbv1.UserSecretName(cluster), Namespace: cluster.Namespace}, &usersSecret)
+	if err != nil {
+		return creds, errors.Wrap(err, "get secret")
+	}
+
+	switch role {
+	case psmdbv1.RoleDatabaseAdmin:
+		creds.Username = string(usersSecret.Data[psmdbv1.EnvMongoDBDatabaseAdminUser])
+		creds.Password = string(usersSecret.Data[psmdbv1.EnvMongoDBDatabaseAdminPassword])
+	case psmdbv1.RoleClusterAdmin:
+		creds.Username = string(usersSecret.Data[psmdbv1.EnvMongoDBClusterAdminUser])
+		creds.Password = string(usersSecret.Data[psmdbv1.EnvMongoDBClusterAdminPassword])
+	case psmdbv1.RoleUserAdmin:
+		creds.Username = string(usersSecret.Data[psmdbv1.EnvMongoDBUserAdminUser])
+		creds.Password = string(usersSecret.Data[psmdbv1.EnvMongoDBUserAdminPassword])
+	case psmdbv1.RoleClusterMonitor:
+		creds.Username = string(usersSecret.Data[psmdbv1.EnvMongoDBClusterMonitorUser])
+		creds.Password = string(usersSecret.Data[psmdbv1.EnvMongoDBClusterMonitorPassword])
+	case psmdbv1.RoleBackup:
+		creds.Username = string(usersSecret.Data[psmdbv1.EnvMongoDBBackupUser])
+		creds.Password = string(usersSecret.Data[psmdbv1.EnvMongoDBBackupPassword])
+	default:
+		return creds, errors.Errorf("not implemented for role: %s", role)
+	}
+
+	return creds, nil
+}
+
+func (r *ReconcilePerconaServerMongoDBRestore) runMongosh(ctx context.Context, cluster *psmdbv1.PerconaServerMongoDB, pod *corev1.Pod, eval string) (*bytes.Buffer, *bytes.Buffer, error) {
```

Comment:
Can we have `cr` instead of `cluster` for `psmdbv1.PerconaServerMongoDB`, it is like a convention for us. Applies to other places as well.

Reply:
no, this is restore controller. `cr` here refers to `PerconaServerMongoDBRestore`